appinventor/appengine/src/com/google/appinventor/client/boxes/BlockSelectorBox.java

```diff
@@ -76,7 +76,9 @@ public void delete() {
   // Starts out not visible. call setVisible(true) to make it visible
   private static final BlockSelectorBox INSTANCE = new BlockSelectorBox();
 
-  private static final Set<String> BUILTIN_DRAWER_NAMES = new HashSet<String>(Arrays.asList("Control", "Logic", "Math", "Text", "Lists", "Colors", "Variables", "Procedures"));
+  private static final Set<String> BUILTIN_DRAWER_NAMES = new HashSet<String>(
```

Comment:
Can't this be made into an enum?

Reply:
@pavi2410 While normally that makes a lot of sense to do that, there are two possible reasons against.

1. Because these get passed via JSNI to be used for filtering blocks in JavaScript, it makes more sense to just keep them as strings (otherwise you'd just have to call `toString()` prior to the JSNI call).
2. We have some experimental editors that add/remove categories as well so this enum would have to encompass all block types across all editors whereas using strings is a bit more flexible.

An alternative approach might be to make the names of the categories constant and reference the constants rather than hard-coding the strings all over the place. That would reduce the chances of typos causing problems (which would also be a benefit of enums).